### ISSUE TITLE
Built-in laptop microphone doesn't work in 16.9.x and Zenith

### ISSUE THREAD

**Issue by Florisv1** [BlissRoms-x86/support#38]
### Description

Built in microphone of laptop is not recognized in Blissos 16.9.x and Blissos 16.9.x Zenith.

### Reproduction steps

```bash
I tried multiple audio recording apps and voice search inside the Google play store, but the built-in Microphone of my laptop is not recognized.

Everything else works in Blissos 16.9.x and 16.9.x Zenith: Bluetooth works, wifi works, my laptop speakers work, the built-in webcam of my laptop works.

My laptop is Lenovo Yoga 7 16IAH7 (82UF0054MH)
My built-in microphone is an: Microphonematrix (Intel Smart Sound Technology for digital microphones).
```

### Category

Sound

### OS Version

16.x

**Comment by Florisv1** [comment]
Are there any steps I can try to fix this? Should I give more info?

**Comment by Florisv1** [comment]
> I tried putting all the volume levels in Alsamixer to max and disabled auto-mute mode, but I still can't use my microphone.
> 
> My microphone (and sound) works out of the box on a live CD of Linux mint without changing any settings, so it's a Blissos issue and not a Linux issue.
> 
> The soundcard, recognized by Alsamixer (in Blissos and Linux Mint) is a Realtek ALC287.

I used su Alsamixer inside Termux and changed all the sound levels (also all input levels) to their maximum.

**Comment by Florisv1** [comment]
Here are the outputs of `su dmesg `and `su lsmod` files of my Blissos installation and the output of `su lsmod` on my Linux mint installation. I also recorded a wav file with the command `arecord --duration=5 file.wav` on my Blissos installation, but the output wav file contained only noise, no sound.

[dmesg_blissos.txt](https://github.com/BlissRoms-x86/support/files/14472622/dmesg.txt)
[lsmod_blisoss.txt](https://github.com/BlissRoms-x86/support/files/14472623/lsmod_blisoss.txt)
[lsmod_mint.txt](https://github.com/BlissRoms-x86/support/files/14472624/lsmod_mint.txt)
[arecord.wav](https://app.koofr.net/links/a2175619-017c-44e1-68d1-3aa7780784f2)

**Comment by hmtheboy154** [comment]
```
[    5.490367] snd_hda_intel 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if info 0x040100
[    5.490420] snd_hda_intel 0000:00:1f.3: Digital mics found on Skylake+ platform, using SOF driver
```

Do you have `/dev/snd/pcmC0D3c` ?

**Comment by Florisv1** [comment]
Yes, `ls -l /dev/snd` lists the following soundcards:
`total 0
crw-rw---- 1 system audio 116,  12 2024-03-03 18:25 controlC0
crw-rw---- 1 system audio 116,  11 2024-03-03 18:25 hwC0D0
crw-rw---- 1 system audio 116,  10 2024-03-03 18:25 hwC0D2
crw-rw---- 1 system audio 116,   5 2024-03-03 18:25 pcmC0D0c
crw-rw---- 1 system audio 116,   4 2024-03-03 18:25 pcmC0D0p
crw-rw---- 1 system audio 116,   6 2024-03-03 18:25 pcmC0D31p
crw-rw---- 1 system audio 116,   7 2024-03-03 18:25 pcmC0D3p
crw-rw---- 1 system audio 116,   8 2024-03-03 18:25 pcmC0D4p
crw-rw---- 1 system audio 116,   9 2024-03-03 18:25 pcmC0D5p
crw-rw---- 1 system audio 116,   2 2024-03-03 18:25 pcmC0D6c
crw-rw---- 1 system audio 116,   3 2024-03-03 18:25 pcmC0D7c
crw-rw---- 1 system audio 116,  33 2024-03-03 18:25 timer`

**Comment by Florisv1** [comment]
I first replied yes, but the answer to your question is no

**Comment by hmtheboy154** [comment]
```
setprop persist.audio.primary.device_in 6
killall audioserver
```

If not working
```
setprop persist.audio.primary.device_in 7
killall audioserver
```

Remember to push every input in `alsamixer` (the `Capture` part) to 100 and unmute

**Comment by Florisv1** [comment]
I can only use this fix in Zenith. In the other recent version (16.9)  the Kernelsu version is too low for the manager to function properly. So I can't give the necessary sudo rights to termux. How do I update kernelsu? Do you have a link which explains this?

**Comment by Florisv1** [comment]
I'm talking about Blissos 16 beta version

**Comment by Florisv1** [comment]
Blissos 16 beta keeps telling me my kernelsu version is too low, even if I install the latest apk.

**Comment by electrikjesus** [comment]
The latest Bliss OS 19.x images do have a known bug with KernelSU.

**Comment by Florisv1** [comment]
Ok, so it's not possible to install the latest KernelSU in the Bèta version? I will use Blissos Zenith for now if I need to use the microphone. I got my microphone working on the Zenith version. Then we can close this issue, I think?

**Comment by Florisv1** [comment]
I will try the microphone in the beta version if KernelSU works in the beta version in the future

**Comment by electrikjesus** [comment]
The improvement that can be made is to automate it for your device. 

Can you please reply with the results of: 

```
cat /sys/class/dmi/id/uevent
```

**Comment by Florisv1** [comment]
[uevent_output.txt](https://github.com/BlissRoms-x86/support/files/14474568/uevent_output.txt)

**Comment by Florisv1** [comment]
> The latest Bliss OS 16.x images do have a known bug with KernelSU.

With this comment you meant kernelsu is not fixable in Blissos 16.x beta? Not by flashing or copying a new version ?

**Comment by electrikjesus** [comment]
No, it can be fixed with an update

**Comment by Florisv1** [comment]
An update of Blissos you mean? ( Not a kernelsu update)

**Comment by electrikjesus** [comment]
Correct. When the next release of Bliss OS 16.9.x comes out, you can update your current install in order to resolve this KernelSU issue
